File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/70-ClimbingStairs/70-ClimbingStairs.java
// Last updated: 8/4/2026, 7:23:43 PM
class Solution {
    public int climbStairs(int n) {
        int [] dp = new int[n+1];
      return stairs(n , 0 , dp);
        
    }
    public static int stairs(int n , int curr , int dp[]){
        int count = 0;
        if(curr == n){
            return 1;
        }
        if(curr > n){
            return 0;
        }
        if(dp[curr] != 0){
            return dp[curr];
        }
       int w1 =  stairs(n ,curr+1,dp);
       int w2 =  stairs(n ,curr+2,dp);
       dp[curr] = w1 + w2;
       return dp[curr];
    }
}

File: https:/github.com/abhinavtripathi17/DSA-Leetcode.git/70-ClimbingStairs/70-ClimbingStairs.cpp
// Last updated: 8/4/2026, 7:23:39 PM
class Solution {
public:
    int dp[46];
    int climbStairs(int n) {
        for(int i = 0 ; i < 46 ; i++){
            dp[i] = -1;
        }
        return solve(n , 0);
    }
    int solve(int n , int steps){
        if(steps == n){
            return 1;
        }
        if(steps > n){
            return 0;
        }
        if(dp[steps] != -1) return dp[steps];
        int t1 = solve(n , steps+1);
        int t2 = solve(n , steps+2);

        return dp[steps] = t1+t2;
    }
};